SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -46,15 +46,43 @@ void USpatialLoadBalanceEnforcer::OnAuthorityIntentComponentUpdated(const Worker
 	}
 }
 
+// This is called whenever this worker becomes authoritative for the ACL component on an entity.
+// It is now this worker's responsibility to make sure that the ACL reflects the current intent,
+// which may have been received before this call.
 void USpatialLoadBalanceEnforcer::AuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
 {
 	if (AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID &&
 		AuthOp.authority == WORKER_AUTHORITY_AUTHORITATIVE)
 	{
+		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(AuthOp.entity_id);
+		if (AuthorityIntentComponent == nullptr)
+		{
+			// TODO(zoning): There are still some entities being created without an authority intent component.
+			// For example, the Unreal created worker entities don't have one. Even though those won't be able to
+			// transition, we should have the intent component on them for completeness.
+		 	UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Requested authority change for entity without AuthorityIntent component. EntityId: %lld"), AuthOp.entity_id);
```

Comment:
Is it supposed to be a Warning or a Log?

Reply:
It shouldn't be a warning imo since there will be snapshot entities without the AuthorityIntent (which is fine) and a warning would be misleading? Personally I think the log statement here should make it a bit clearer that it's fine OR we change the logic to more explicitly check the entity ID doesn't match the known entity IDs in the snapshot then upgrade to a warning?

Reply:
I had been thinking that any entity which could be authoritative on an UnrealWorker should have an AuthorityIntent, so from that perspective it could be a Warning, but I was also hoping we could go even farther to making it an Error (except there are some right now, so we can't yet)

Reply:
After having a 2nd search, looks to me like most places actually set the AuthorityIntent on entities spawned (or if they don't then there's a check for bEnableLoadBalancer and this code will never be reached if that value isn't true). The only place I found that definitely doesn't have one is `void USpatialSender::CreateServerWorkerEntity(int AttemptCounter)` which I don't think any workers have interest in & shouldn't follow the `HandleActorAuthority` code path - which places did you think didn't add the component? Can believe I definitely missed something 😄Outside of that, I think it should be fine to upgrade to error 👍